### PR TITLE
feat: update docs

### PR DESCRIPTION
**Pull request by loicsay**

### CODE REVIEW

File: pkg/cmd/profile/add/add.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func NewAddCmd(f *cmdutil.Factory, runF func(*AddOptions) error) *cobra.Command
 			# Add a new profile (interactive)
 			$ algolia profile add
 
-			# Add a new profile (non-interactive)
+			# Add a new profile (non-interactive) and set to default
```
Comment (clemfromspace) [inline pkg/cmd/profile/add/add.go:43]:
```suggestion
			# Add a new profile (non-interactive) and set it as default
```

File: pkg/cmd/objects/browse/browse.go [hunk 2]
```diff
@@ -43,8 +43,17 @@ func NewBrowseCmd(f *cmdutil.Factory) *cobra.Command {
 			This command browse the objects of the specified index.
 		`),
 		Example: heredoc.Doc(`
-			# Browse the objects from the "TEST_PRODUCTS_1" index
-			$ algolia objects browse TEST_PRODUCTS_1
+			# Browse the objects from the "BOOKS" index
+			$ algolia objects browse BOOKS
+
+			# Select which attributes to retrieve
```
Comment (clemfromspace) [inline pkg/cmd/objects/browse/browse.go:49]:
```suggestion
			# Browse the objects from the "BOOKS" index and select which attributes to retrieve
```

File: pkg/cmd/objects/browse/browse.go [hunk 3]
```diff
@@ -43,8 +43,17 @@ func NewBrowseCmd(f *cmdutil.Factory) *cobra.Command {
...
+			$ algolia objects browse BOOKS --attributesToRetrieve author,title,description
+
+			# Browse with filters
+			& algolia search BOOKS --filters "'(category:Book OR category:Ebook) AND _tags:published'"
```
Comment (clemfromspace) [inline pkg/cmd/objects/browse/browse.go:53]:
```suggestion
			$ algolia search BOOKS --filters "'(category:Book OR category:Ebook) AND _tags:published'"
```

File: pkg/cmd/objects/browse/browse.go [hunk 4]
```diff
@@ -43,8 +43,17 @@ func NewBrowseCmd(f *cmdutil.Factory) *cobra.Command {
...
 		Example: heredoc.Doc(`
-			# Browse the objects from the "TEST_PRODUCTS_1" index
-			$ algolia objects browse TEST_PRODUCTS_1
+			# Browse the objects from the "BOOKS" index
+			$ algolia objects browse BOOKS
+
+			# Select which attributes to retrieve
+			$ algolia objects browse BOOKS --attributesToRetrieve author,title,description
+
+			# Browse with filters
```
Comment (clemfromspace) [inline pkg/cmd/objects/browse/browse.go:52]:
```suggestion
			# Browse the objects from the "BOOKS" index with filters
```

File: pkg/cmd/objects/browse/browse.go [hunk 5]
```diff
@@ -43,8 +43,17 @@ func NewBrowseCmd(f *cmdutil.Factory) *cobra.Command {
...
+			& algolia search BOOKS --filters "'(category:Book OR category:Ebook) AND _tags:published'"
+
+			# Browse with custom pagination
```
Comment (clemfromspace) [inline pkg/cmd/objects/browse/browse.go:55]:
```suggestion
			# Browse the objects from the "BOOKS" index with custom pagination
```

File: pkg/cmd/objects/browse/browse.go [hunk 6]
```diff
@@ -43,8 +43,17 @@ func NewBrowseCmd(f *cmdutil.Factory) *cobra.Command {
...
+			& algolia search BOOKS --hitsPerPage 10 --page 3
```
Comment (clemfromspace) [inline pkg/cmd/objects/browse/browse.go:56]:
```suggestion
			$ algolia search BOOKS --hitsPerPage 10 --page 3
```

File: pkg/cmd/search/search.go [hunk 7]
```diff
@@ -42,7 +42,16 @@ func NewSearchCmd(f *cmdutil.Factory) *cobra.Command {
 		Long:              `Search for objects in your index.`,
 		Example: heredoc.Doc(`
 			# Search for objects in your index
```
Comment (clemfromspace) [inline pkg/cmd/search/search.go:44]:
```suggestion
			# Search for objects in the "BOOKS" index matching the query "tokien"
```

File: pkg/cmd/search/search.go [hunk 8]
```diff
@@ -42,7 +42,16 @@ func NewSearchCmd(f *cmdutil.Factory) *cobra.Command {
...
-			$ algolia search PRODUCTS --query "foo"
+			$ algolia search BOOKS --query "tolkien"
+
+			# Search with filters
+			& algolia search BOOKS --query "tolkien" --filters "'(category:Book OR category:Ebook) AND _tags:published'"
```
Comment (clemfromspace) [inline pkg/cmd/search/search.go:48]:
```suggestion
			$ algolia search BOOKS --query "tolkien" --filters "'(category:Book OR category:Ebook) AND _tags:published'"
```

File: pkg/cmd/search/search.go [hunk 9]
```diff
@@ -42,7 +42,16 @@ func NewSearchCmd(f *cmdutil.Factory) *cobra.Command {
 		Long:              `Search for objects in your index.`,
 		Example: heredoc.Doc(`
 			# Search for objects in your index
-			$ algolia search PRODUCTS --query "foo"
+			$ algolia search BOOKS --query "tolkien"
+
+			# Search with filters
```
Comment (clemfromspace) [inline pkg/cmd/search/search.go:47]:
```suggestion
			# Search for objects in the "BOOKS" index matching the query "tokien" with filters
```

File: pkg/cmd/search/search.go [hunk 10]
```diff
@@ -42,7 +42,16 @@ func NewSearchCmd(f *cmdutil.Factory) *cobra.Command {
...
+			& algolia search BOOKS --query "tolkien" --filters "'(category:Book OR category:Ebook) AND _tags:published'"
+
+			# Setting number of hits per page and specify the page to retrieve
```
Comment (clemfromspace) [inline pkg/cmd/search/search.go:50]:
```suggestion
			# Search for objects in the "BOOKS" index matching the query "tokien" while setting the number of hits per page and specifying the page to retrieve
```

File: pkg/cmd/search/search.go [hunk 11]
```diff
@@ -42,7 +42,16 @@ func NewSearchCmd(f *cmdutil.Factory) *cobra.Command {
...
+			$ algolia search BOOKS --query "tolkien" --hitsPerPage 2 --page 4
+
+			# Export result to json file
```
Comment (clemfromspace) [inline pkg/cmd/search/search.go:53]:
```suggestion
			# Search for objects in the "BOOKS" index matching the query "tolkien" and export the results to a new line delimited JSON (ndjson) file
```

File: pkg/cmd/search/search.go [hunk 12]
```diff
@@ -42,7 +42,16 @@ func NewSearchCmd(f *cmdutil.Factory) *cobra.Command {
...
+			$ algolia search BOOKS --query "tolkien" > books.json
```
Comment (clemfromspace) [inline pkg/cmd/search/search.go:54]:
```suggestion
			$ algolia search BOOKS --query "tolkien" > books.ndjson
```

**Review by clemfromspace — CHANGES_REQUESTED** [review]
Thank you for working on this!
I added a few suggestions because I am thinking that we should document each of the examples as "independant" ones. IMHO, we should not assume that the context for a given example is provided by the previous example.

Let me know if that's clear!

**Comment by loicsay** [comment]
@clemfromspace You're right! I'll keep that in mind for the future, context should always be explicit for each case
Thanks 🙏

File: pkg/cmd/objects/browse/browse.go [hunk 13]
```diff
@@ -43,8 +43,17 @@ func NewBrowseCmd(f *cmdutil.Factory) *cobra.Command {
 			This command browse the objects of the specified index.
 		`),
 		Example: heredoc.Doc(`
```
Comment (kai687) [inline pkg/cmd/objects/browse/browse.go:45]: Can we add an example for 'exporting the objects to a file':

`algolia objects browse BOOKS > books.ndjson`

File: pkg/cmd/objects/browse/browse.go [hunk 14]
```diff
@@ -43,8 +43,17 @@ func NewBrowseCmd(f *cmdutil.Factory) *cobra.Command {
...
-			# Browse the objects from the "TEST_PRODUCTS_1" index
-			$ algolia objects browse TEST_PRODUCTS_1
+			# Browse the objects from the "BOOKS" index
+			$ algolia objects browse BOOKS
+
+			# Browse the objects from the "BOOKS" index and select which attributes to retrieve
+			$ algolia objects browse BOOKS --attributesToRetrieve author,title,description
+
+			# Browse the objects from the "BOOKS" index with filters
+			$ algolia search BOOKS --filters "'(category:Book OR category:Ebook) AND _tags:published'"
```
Comment (kai687) [inline pkg/cmd/objects/browse/browse.go:53]:
```suggestion
			$ algolia search BOOKS --filters "'(category:Book OR category:Ebook) AND _tags:published'"
```

Should this not be using the `browse` command?

File: pkg/cmd/objects/browse/browse.go [hunk 15]
```diff
@@ -43,8 +43,17 @@ func NewBrowseCmd(f *cmdutil.Factory) *cobra.Command {
...
+
+			# Browse the objects from the "BOOKS" index with custom pagination
+			$ algolia search BOOKS --hitsPerPage 10 --page 3
```
Comment (kai687) [inline pkg/cmd/objects/browse/browse.go:56]:
```suggestion
			$ algolia search BOOKS --hitsPerPage 10 --page 3
```

I think `browse` doesn't use pagination.

File: pkg/cmd/objects/browse/browse.go (hunk 14 above)
Comment (loicsay) [reply pkg/cmd/objects/browse/browse.go:53]: You're right, I forgot to change it after copy/pasting 🙏

File: pkg/cmd/objects/browse/browse.go (hunk 15 above)
Comment (loicsay) [reply pkg/cmd/objects/browse/browse.go:56]: You're right, I tried and it doesn't work. However it's written in the doc: https://www.algolia.com/doc/tools/cli/commands/algolia-objects/#algolia-objects-browse-pagination

Should we remove these options from the doc?

File: pkg/cmd/objects/browse/browse.go (hunk 13 above)
Comment (loicsay) [reply pkg/cmd/objects/browse/browse.go:45]: Sure! 🚀

**Review by clemfromspace — APPROVED** [review]

**Pull request merged**